ocaml/xapi/xapi_vm_snapshot.ml

```diff
@@ -169,27 +169,23 @@ let checkpoint ~__context ~vm ~new_name =
 
 (* The following code have to run on the master as it manipulates the DB cache directly. *)
 let copy_vm_fields ~__context ~metadata ~dst ~do_not_copy ~overrides =
-  if not (Pool_role.is_master ()) then
-    raise
-      Api_errors.(
-        Server_error
-          ( internal_error
-          , ["copy_vm_fields: Aborting because the host is not master"]
-          )
-      ) ;
+  ( if not (Pool_role.is_master ()) then
+      let msg = "copy_vm_fields: Aborting because the host is not master" in
+      raise Api_errors.(Server_error (internal_error, [msg]))
+  ) ;
   debug "copying metadata into %s" (Ref.string_of dst) ;
   let db = Context.database_of __context in
   let module DB = (val Db_cache.get db : Db_interface.DB_ACCESS) in
   List.iter
     (fun (key, value) ->
-      let value =
-        if List.mem_assoc key overrides then
-          List.assoc key overrides
-        else
-          value
-      in
+      let value = Option.value ~default:value (List.assoc_opt key overrides) in
       if not (List.mem key do_not_copy) then
-        DB.write_field db Db_names.vm (Ref.string_of dst) key value
+        try DB.write_field db Db_names.vm (Ref.string_of dst) key value
+        with Db_exn.DBCache_NotFound ("missing column", _, name) ->
```

Comment:
matching exception based on string might be brittle. Perhaps one compromise would be to define a constant string in schema.ml for "missing column" use that in the raise, and then use it again here.
Or turn that string into a variant.